float64 fieldX
float64 fieldY